cloud_functions/tsne_embedding.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Used to create a tsne embedding coordinates from the given embeddings.
+This Cloud Function runs to create coordinates of all the embeddings that have
+been uploaded to the bucket. The function checks to make sure all the embedding
+s have been uploaded, then gets the coordinates.
+"""
+import numpy as np
+import os
+from file_path_utils import get_user_and_dataset_name
+from sklearn.manifold import TSNE
+from google.cloud import storage, datastore
+import json
+import tempfile
+
+# Cloud APIs:
+storage_client = storage.Client()
+datastore_client = datastore.Client('step-2020-johndallard')
+OUTPUT_BUCKET_NAME = 'spritesheet_json'
+
+
+def run_tsne(trigger_file, _):
+    """
+        Trigger function to run. Gets t-SNE coordinates from
+        embeddings.
+        First checks to make sure all embeddings are present, and
+        after doing so saves all of them to an array. Once they have
+        been saved, the points are saved from the tsne_embed function
+        and saved to a json file in an output bucket
+        Args:
+            trigger_file: 
+                Dictionary that contains data specific to the event.
+            _:
+                Cloud functions event metadata.
+    """
+
+    # Looks for all embeddings in the embedding folder to ensure they are all t
+    # here
+    folder = '/'.join(trigger_file['name'].split('/')[:-1])
```

Comment:
Can you fix this as well? Here and in all other files in other PRs.

Reply:
yup